src/libs/SidebarUtils.js

```diff
@@ -246,7 +246,7 @@ function getOptionData(reportID) {
     if (ReportUtils.isReportMessageAttachment({text: report.lastMessageText, html: report.lastMessageHtml})) {
         lastMessageTextFromReport = `[${Localize.translateLocal('common.attachment')}]`;
     } else {
-        lastMessageTextFromReport = Str.htmlDecode(report ? report.lastMessageText : '');
+        lastMessageTextFromReport = report ? report.lastMessageText : '';
```

Comment:
This change will cause following regression.
<img width="1421" alt="Screenshot 2023-02-22 at 21 58 10" src="https://user-images.githubusercontent.com/25876548/220686700-671a5956-6fcf-4e28-8b48-cfb097581c94.png">

Reply:
@sobitneupane Thanks for pointing that out. Firstly, I will remove the html.decode in the title of the notification (it is the user name) 

We are having 2 cases with the message (it includes the message in chat, the last message in LHN, and the new message in the notification):
Case 1:  we use htmlDecode and we send a request money from the user whose name is &nbsp. The BE will return `request 111 from &nbsp`. It displays `request 111 from &` in LHN, IOU, and the notification

Case 2: we don't use htmlDecode the actions in case 1 will be resolved but if we send a normal text as you did above: `hello &#38`, The BE will return `&amp;#38` then It displays incorrectly in LHN 

To sync all cases, I suggest that we should update the request money message returned from BE: `request 111 from &nbsp` -> `request 111 from &amp;nbsp` and we still use html.decode in all messages. 
We only remove the html.decode in the title of the username, not in the message

Reply:
@tienifr That sounds like a plan. 

> To sync all cases, I suggest that we should update the request money message returned from BE: request 111 from &nbsp -> request 111 from &amp;nbsp and we still use html.decode in all messages.
We only remove the html.decode in the title of the username, not in the message

Is there any chance, we can handle it in front end? Like are we feeding backend with the username from the front or is it being handled in backend completely?

Reply:
@sobitneupane I'm afraid that it's handled in backend completely, you can check the code here
https://github.com/Expensify/App/blob/94057baaf27d1f6c28277fc5cd713ee42ec10ada/src/libs/actions/IOU.js#L196-L207. 
I thought about handling it in frontend. I can detect the message is IOU in report screen, but not in LHN and browser notification. What is your thought?

Reply:
@pecanoro 
@tienifr is proposing [some changes](https://github.com/Expensify/App/pull/15340#discussion_r1115241821) in backend here. What are your thoughts on this?

Reply:
If I am following, are you proposing to encode IOU messages in the back-end so the front-end can decode it properly?

Reply:
@pecanoro That's right, like we do with the normal message

Reply:
Ok, let me check what I need to change in the back-end

Reply:
Found where we send the message and I will create an internal PR. Just one question, @sobitneupane where in the image is the regression? I am blind and I can't see what's wrong, but maybe because all HTML entities get blurred together haha

Reply:
PR created, I will keep you both posted when it gets deployed.

Reply:
@pecanoro `hello &#38` is sent in the message, `hello &amp;#38` is displayed as lastMessageText in LHN.

Reply:
@sobitneupane @pecanoro Thanks for your help? Should we hold the issue until the internal PR is deployed?

Reply:
@sobitneupane Do you mean we have this similar back-end bug for the message in the LHN? I only fixed the bug for IOU actions. It should be merged today and deploy tomorrow (hopefully)

Reply:
> Do you mean we have this similar back-end bug for the message in the LHN? 
@pecanoro No. The message returned are already HTML encoded. The IOU actions were not html encoded before. So, after your change in IOU actions, both message and IOU actions should be consistent and thus, can be handled easily in frontend.

Reply:
@tienifr Looks like the change is already deployed. Let's wrap it up as soon as possible.

We are getting many IOU Actions related bug reports:

https://expensify.slack.com/archives/C049HHMV9SM/p1677580450325309
https://expensify.slack.com/archives/C049HHMV9SM/p1677571343924769
https://expensify.slack.com/archives/C049HHMV9SM/p1677560414261149

Please do make sure to verify that the changes in this PR solve above bugs.

Reply:
Yes, changes were deployed to staging last night! Let's make sure we get this merged asap 😄

Reply:
I forgot to warn our QA vendor and they thought the back-end changes were a deploy blocker, so we CPed a [fix](https://github.com/Expensify/App/pull/15578) for the message, don't forget to merge with main to get the last changes.

Reply:
@sobitneupane All bugs you mentioned in [here](https://github.com/Expensify/App/pull/15340#discussion_r1119881875) is the same and [this PR](https://github.com/Expensify/App/pull/15578) fixed it I have tested again and it have worked correctly

<img width="1440" alt="Screen Shot 2023-03-01 at 08 42 45" src="https://user-images.githubusercontent.com/113963320/222024170-18027d52-a0c5-4444-af2c-18c1afa213de.png">

cc @pecanoro